src/EventInterface.php

```diff
@@ -50,23 +55,29 @@ public function setName($name);
     /**
      * Set the event target/context
      *
-     * @param  null|string|object $target
+     * @param object|string|null $target
+     * @template NewTTarget of object|string|null
+     * @psalm-param NewTTarget $target
+     * @psalm-this-out self<NewTTarget, TParams>
```

Comment:
Will psalm-this-out preserve the child class type? Just want to be sure that's verified